include/boost/date_time/posix_time/conversion.hpp

```diff
@@ -26,7 +26,7 @@ namespace posix_time {
   inline
   ptime from_time_t(std::time_t t)
   {
-    return ptime(gregorian::date(1970,1,1)) + seconds(static_cast<long>(t));
+    return ptime(gregorian::date(1970,1,1)) + seconds(static_cast<boost::int64_t>(t));
```

Comment:
See alternative suggestion in https://github.com/boostorg/date_time/issues/87 - is the cast necessary here?